sdk/identity/Azure.Identity/src/KubernetesProxyHttpHandler.cs

```diff
@@ -0,0 +1,265 @@
+
+            // Copy headers from HttpRequestMessage to Request
+            foreach (var header in request.Headers)
+            {
+                foreach (var value in header.Value)
+                {
+                    pipelineRequest.Headers.Add(header.Key, value);
+                }
+            }
+
+            // Copy content if present
+            if (request.Content != null)
+            {
+                foreach (var contentHeader in request.Content.Headers)
+                {
+                    foreach (var value in contentHeader.Value)
+                    {
+                        pipelineRequest.Headers.Add(contentHeader.Key, value);
+                    }
+                }
+                var contentStream = await request.Content.ReadAsStreamAsync().ConfigureAwait(false);
+                pipelineRequest.Content = RequestContent.Create(contentStream);
+            }
+
+            Response pipelineResponse = await _pipeline.SendRequestAsync(pipelineRequest, cancellationToken).ConfigureAwait(false);
+
+            // Convert Response back to HttpResponseMessage
+            var response = new HttpResponseMessage((System.Net.HttpStatusCode)pipelineResponse.Status);
+            response.ReasonPhrase = pipelineResponse.ReasonPhrase;
+
+            // Copy response content
+            if (pipelineResponse.ContentStream != null)
+            {
+                response.Content = new StreamContent(pipelineResponse.ContentStream);
+            }
+
+            // Copy response headers
+            foreach (var header in pipelineResponse.Headers)
+            {
+                if (pipelineResponse.Headers.TryGetValues(header.Name, out var values))
+                {
+                    if (!response.Headers.TryAddWithoutValidation(header.Name, values))
+                    {
+                        response.Content?.Headers.TryAddWithoutValidation(header.Name, values);
+                    }
+                }
+            }
+
+            return response;
+        }
+
+        private void RewriteRequestUrl(HttpRequestMessage request)
+        {
+            if (request.RequestUri == null)
+            {
+                throw new InvalidOperationException("Request URI is null.");
+            }
+
+            string pathAndQuery = request.RequestUri.PathAndQuery;
+            UriBuilder newUri = new UriBuilder(_config.ProxyUrl);
+
+            string proxyPath = newUri.Path.TrimEnd('/');
+            string requestPath = pathAndQuery.TrimStart('/');
+
+            string combinedPath;
+            if (string.IsNullOrEmpty(proxyPath) || proxyPath == "/")
+            {
+                combinedPath = "/" + requestPath;
+            }
+            else
+            {
+                combinedPath = proxyPath + "/" + requestPath;
+            }
+
+            // Build URI string directly to avoid UriBuilder.Path escaping
+            string uriString = $"{newUri.Scheme}://{newUri.Host}:{newUri.Port}{combinedPath}";
```

Comment:
does `newUri.Port` always have a value, even when `AZURE_KUBERNETES_TOKEN_PROXY_URL` is something like `https://proxy.url/token`? Wondering whether we could get `https://proxy.url:/token` here and if so, whether the HTTP client would accept that

Reply:
Yes, `newUri.Port` always has a value because it defaults to 443 or 80, depending if it's HTTPS or HTTP.